cli/cmd/middleware.go

```diff
@@ -29,6 +59,28 @@ func WithResultHandler(runFn RunFn) CobraRunFn {
 	}
 }
 
+func handleError(ctx context.Context, err error) {
+	reqErr := resourcemanager.RequestError{}
+	if errors.As(err, &reqErr) && reqErr.IsAuthError {
+		handleAuthError(ctx)
+	} else {
+		OnError(err)
+	}
+}
+
+func handleAuthError(ctx context.Context) {
+	ui.DefaultUI.Warning("Your authentication token has expired, please log in again.")
```

Comment:
Does this handles environment tokens expiration?

Reply:
That's a good question. I didn't test that case, but this mechanism is triggered whenever the server returns a 401. So if an expired token throws a 401, then this will be triggered.